projects/coordinator/src/main/java/org/batfish/coordinator/CrossDomainFilter.java

```diff
@@ -1,13 +1,33 @@
 package org.batfish.coordinator;
 
+import javax.annotation.Priority;
+import javax.ws.rs.HttpMethod;
 import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
 import javax.ws.rs.container.ContainerResponseContext;
 import javax.ws.rs.container.ContainerResponseFilter;
+import javax.ws.rs.core.CacheControl;
+import javax.ws.rs.core.Response;
 import javax.ws.rs.ext.Provider;
 
 /** Allow the system to serve xhr level 2 from all cross domain site */
 @Provider
-public class CrossDomainFilter implements ContainerResponseFilter {
+@Priority(-1) /* Highest priority to set the CORS filter as the first one to run. */
+public class CrossDomainFilter implements ContainerRequestFilter, ContainerResponseFilter {
+  /**
+   * Recognizes a CORS preflight request, and return OK without any further downstring processing in
+   * such a case.
+   */
+  @Override
+  public void filter(ContainerRequestContext creq) {
+    if (HttpMethod.OPTIONS.equals(creq.getMethod())) {
+      // Cache these responses.
+      CacheControl control = new CacheControl();
+      control.setMaxAge(3600);
+      creq.abortWith(Response.ok().cacheControl(control).build());
```

Comment:
Won't we need a header "Access-Control-Allow-Methods" in the response, listing the supported HTTP methods, apart from the OK response ? [https://developer.mozilla.org/en-US/docs/Glossary/Preflight_request](reference)